packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -292,8 +293,20 @@ export class BaseInvocationScope<TReturn = any> {
     assert(this.program.account, 'Wallet is required!');
 
     const transactionRequest = await this.getTransactionRequest();
+    const { minGasPrice } = this.getProvider().getGasConfig();
+    const { maxFee, gasUsed } = await this.getTransactionCost();
+
+    const specifiedGasLimit = this.txParameters?.gasLimit || this.hasCallParamsGasLimit;
+    const specifiedGasPrice = this.txParameters?.gasPrice;
+
+    // set defaults for gasLimit and gasPrice if not specified
+    if (!specifiedGasLimit) {
+      transactionRequest.gasLimit = gasUsed;
+    }
 
-    const { maxFee } = await this.getTransactionCost();
+    if (!specifiedGasPrice) {
+      transactionRequest.gasPrice = minGasPrice;
+    }
```

Comment:
This whole block seems repeated 3x times in this file.

Should we DRY it by encapsulating it in a new function or something?

Reply:
Yessir: https://github.com/FuelLabs/fuels-ts/pull/1708/commits/8f95d7b7670bb2c16cd01da1486029dfc45697ec